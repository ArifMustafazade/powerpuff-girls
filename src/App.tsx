import React from 'react';
import { BrowserRouter, Route } from 'react-router-dom';

// containers
import { LandingPage, EpisodeDetails } from './containers';

/** @App root */
const App = (): JSX.Element => {
    return (
        <BrowserRouter>
            <div className="app">
                <header className="App-header">Header</header>
                <Route path="/" exact component={LandingPage} />
                <Route path="/episodes/:id" component={EpisodeDetails} />
            </div>
        </BrowserRouter>
    );
};

export default App;
