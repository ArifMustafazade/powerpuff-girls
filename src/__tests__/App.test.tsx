import React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';

// components
import App from '../App';

/** tests */
describe('<App />', () => {
    let wrapper: ShallowWrapper<any>;

    beforeEach(() => {
        wrapper = shallow(<App />);
    })

    test('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should render with correct title', () => {
        expect(wrapper.find('.App-header').text()).toBe('Hello');
    });

})



