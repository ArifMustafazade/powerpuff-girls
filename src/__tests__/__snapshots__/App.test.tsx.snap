// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<BrowserRouter>
  <div
    className="app"
  >
    <header
      className="App-header"
    >
      Header
    </header>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/episodes/:id"
    />
  </div>
</BrowserRouter>
`;
